config/config-server/resources/cruise-config.xsd

```diff
@@ -712,7 +712,7 @@
     </xsd:simpleType>
     <xsd:simpleType name="labelType">
         <xsd:restriction base="xsd:string">
-            <xsd:pattern value="(([a-zA-Z0-9_\-.!~*'()#:])*[$#]\{[a-zA-Z0-9_\-.!~*'()#:]+\}([a-zA-Z0-9_\-.!~*'()#:])*)+"/>
+            <xsd:pattern value="(([a-zA-Z0-9_\-.!~*'()#:])*[$#]\{[a-zA-Z0-9_\-.!~*'()#:]+(\[:(\d+)\])?\}([a-zA-Z0-9_\-.!~*'()#:])*)+"/>
```

Comment:
you might want to add a config load test. something like [here](https://github.com/gocd/gocd/blob/4b838fed09498c01b0feb876080de7b74f9f18a8/config/config-server/test/com/thoughtworks/go/config/MagicalGoConfigXmlLoaderTest.java#L1082)